### DISCUSSION TITLE
How can I stop Argo Events hooked up with PubSub from creating non-existant topics/subscriptions?

### DISCUSSION THREAD

**Discussion by ghost** [argoproj/argo-events#1096] [Q&A] [answered]
Hello!

I'm using Argo Events with PubSub. The corresponding topic and subscription is supposed to be maintained only by Terraform. Technically speaking I'd like a loose coupling between Argo Events and PubSub. And practically speaking I don't want Argo Events to create resources (topic and subscription). Because this will cause an "Resource already exists in the project" error in some scenarios. Is it possible to keep Argo Events from creating topics/subscriptions by configuration?

Let's say I have the situation where I need to tear down my TF stack (including the PubSub stuff used by Argo Events) - then I need to prevent (somehow) Argo Events from creating those because if it does I cannot redeploy as states are brought out of sync. Tf thinks the PubSub stuff doesn't exist anymore but Argo Events in the meantime sets it up again. :/

Thanks

**Comment by whynowy** [comment, marked as answer]
There's not a switch in Argo Events right now to control that, but could you achieve that by limiting the privileges? Say you don't grant the permission of creating topics and subscriptions.

**Reply by ghost** [reply]
Of course - that makes sense! I think I prematureley discared this option because expected complaints from Argo Events.
